packages/flutter_tools/lib/src/android/aar.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../base/common.dart';
+import '../build_info.dart';
+import '../project.dart';
+
+import 'android_sdk.dart';
+import 'gradle.dart';
+
+/// Builds the AAR and POM files for the current Flutter module or plugin.
+Future<void> buildAar({
+  @required FlutterProject project,
+  @required AndroidBuildInfo androidBuildInfo,
+  @required String target,
+  @required String outputDir,
+}) async {
+  if (!project.android.isUsingGradle) {
+    throwToolExit(
+        'The build process for Android has changed, and the current project configuration\n'
+            'is no longer valid. Please consult\n\n'
+            '  https://github.com/flutter/flutter/wiki/Upgrading-Flutter-projects-to-build-with-gradle\n\n'
+            'for details on how to upgrade the project.'
+    );
+  }
+  if (!project.manifest.isModule && !project.manifest.isPlugin) {
+    throwToolExit('AARs can only be built for plugin or module projects.');
```

Comment:
Feel free to leave a bug and move on, but we'll probably get questions about "I accidentally `flutter create`ed an app type project and now I want to add-to-app but don't know how to proceed" at some point.

Reply:
I think at that point we just ditch the module project entirely and have this command or some other know how to generate the right artifacts to build a library project.

But definitely for another time.

Reply:
sgtm.